pkg/cmd/factory/default.go

```diff
@@ -113,6 +113,11 @@ func httpClientFunc(f *cmdutil.Factory, appVersion string) func() (*http.Client,
 func newGitClient(f *cmdutil.Factory) *git.Client {
 	io := f.IOStreams
 	ghPath := f.Executable()
+	stdout, ok := io.Out.(*os.File)
+	if ok && iostreams.IsCygwinTerminal(stdout.Fd()) {
```

Comment:
If stdout was redirected, then I think this would result in a false negative. So an invocation in cygwin like `gh repo clone ... >/dev/null` could start erroring out again due to the original bug.

The go-gh term package has an approach of inspecting the commanding terminal even when inputs and outputs were redirected: https://github.com/cli/go-gh/blob/86f3a6e29984a9cbbef8f8d8ce031a6d43739e3d/pkg/term/env.go#L122-L125